lib/block_view_lockups.go

```diff
@@ -0,0 +1,2178 @@
+			txn.PublicKey,
+			txMeta.ProfilePublicKey.ToBytes(),
+			true)
+		if transactorBalanceEntry == nil || transactorBalanceEntry.isDeleted {
+			return 0, 0, nil,
+				errors.Wrapf(RuleErrorCoinLockupBalanceEntryDoesNotExist, "_connectCoinLockup")
+		}
+
+		// Validate the balance entry as having sufficient funds.
+		transactorBalanceNanos256 = transactorBalanceEntry.BalanceNanos.Clone()
+		if txMeta.LockupAmountBaseUnits.Gt(transactorBalanceNanos256) {
+			return 0, 0, nil,
+				errors.Wrap(RuleErrorCoinLockupInsufficientCoins, "_connectCoinLockup")
+		}
+
+		// We store the previous transactor balance entry in the event we need to revert the transaction.
+		prevTransactorBalanceEntry = transactorBalanceEntry.Copy()
+
+		// Spend the transactor's DAO coin balance.
+		transactorBalanceEntry.BalanceNanos =
+			*uint256.NewInt().Sub(&transactorBalanceEntry.BalanceNanos, txMeta.LockupAmountBaseUnits)
+		bav._setDAOCoinBalanceEntryMappings(transactorBalanceEntry)
+
+		// Create a copy of the associated CoinEntry in the event we must roll back the transaction.
+		prevCoinEntry = profileEntry.DAOCoinEntry.Copy()
+
+		// Update CoinsInCirculation and NumberOfHolders associated with the DAO coin balance.
+		profileEntry.DAOCoinEntry.CoinsInCirculationNanos = *uint256.NewInt().Sub(
+			&profileEntry.DAOCoinEntry.CoinsInCirculationNanos,
+			txMeta.LockupAmountBaseUnits)
+		if transactorBalanceEntry.BalanceNanos.IsZero() {
+			profileEntry.DAOCoinEntry.NumberOfHolders--
+		}
+		bav._setProfileEntryMappings(profileEntry)
+	}
+
+	// By now we know the transaction to be valid. We now source yield information from either
+	// the profile's yield curve or the raw DeSo yield curve. Because there's some choice in how
+	// to determine the yield when the lockup duration falls between two profile specified yield curve
+	// points, we return here the two local points and choose/interpolate between them below.
+	leftYieldCurvePoint, rightYieldCurvePoint, err := bav.GetLocalYieldCurvePoints(profilePKID, lockupDurationNanoSeconds)
+	if err != nil {
+		return 0, 0, nil,
+			errors.Wrap(err, "_connectCoinLockup failed to fetch yield curve points")
+	}
+
+	// Here we interpolate (choose) the yield between the two returned local yield curve points.
+	//
+	// If we fall between two points, we choose the left yield curve point (i.e. the one with lesser lockup duration).
+	// The transactor earns yield only for the lockup duration specified by the left yield curve point but will
+	// be unable to unlock the coins until the transaction specified lockup duration expires.
+	txnYieldBasisPoints := uint64(0)
+	txnYieldEarningDurationNanoSecs := int64(0)
+	if leftYieldCurvePoint != nil &&
+		leftYieldCurvePoint.LockupDurationNanoSecs < lockupDurationNanoSeconds {
+		txnYieldBasisPoints = leftYieldCurvePoint.LockupYieldAPYBasisPoints
+		txnYieldEarningDurationNanoSecs = leftYieldCurvePoint.LockupDurationNanoSecs
+	}
+	if rightYieldCurvePoint != nil &&
```

Comment:
You could have avoided this second edge-case if you made the left point <= and the right point strictly >. No need to fix it just saying.